src/main/java/com/theelixrlabs/healthcare/service/DoctorService.java

```diff
@@ -87,17 +110,41 @@ public DoctorDto getDoctorById(UUID doctorId) throws CustomException {
                 .build();
     }
 
+    /**
+     * Deletes a doctor by their id.
+     *
+     * @param doctorId    The string representation of the doctor's UUID.
+     * @return A success message upon successful deletion.
+     * @throws CustomException    If the doctor is not found or is assigned to any patients.
+     */
+    public String deleteDoctorById(String doctorId) throws CustomException {
+        UUID validDoctorId = validator.validateAndConvertToUUID(doctorId, MessageConstants.INVALID_UUID);
+
+        // Check if a doctor with the validDoctorId exists in the repository.
+        if (!doctorRepository.existsById(validDoctorId))
+            throw new ResourceNotFoundException(messageUtil.getMessage(MessageConstants.DOCTOR_ID_NOT_FOUND));
+
+        // Check if the doctor is assigned to any patients before deletion.
+        if (isDoctorAssignedToPatient(validDoctorId)) {
+            throw new CustomException(messageUtil.getMessage(MessageConstants.DOCTOR_DELETION_FAILED_ASSIGNED_TO_PATIENT));
+        }
+
+        // If no patients are assigned to the doctor, proceed with deletion.
+        doctorRepository.deleteById(validDoctorId);
```

Comment:
What about deleting same doctor's un-assignment data from assignment collection ? 
Discuss and update accordingly as per your requirement

Reply:
If I delete the doctor assignments, then some patients might lose their assignments data.